controllers/argocd/deployment.go

```diff
@@ -34,6 +34,28 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+// getArgoCDRepoServerReplicas will return the size value for the argocd-repo-server replica count.
+func getArgoCDRepoServerReplicas(cr *argoprojv1a1.ArgoCD) *int32 {
+	replicas := common.ArgoCDRepoServerReplicas
+	if cr.Spec.Repo.Replicas != nil {
+		if *cr.Spec.Repo.Replicas >= 0 && *cr.Spec.Repo.Replicas != replicas {
+			replicas = *cr.Spec.Repo.Replicas
+		}
+	}
+	return &replicas
+}
+
+// getArgoCDServerReplicas will return the size value for the argocd-server replica count.
```

Comment:
Please see my comment in `getArgoCDRepoServerReplicas()`.  It applies to this function as well.